pkg/client/unversioned/client.go

```diff
@@ -114,6 +115,12 @@ func (c *Client) ComponentStatuses() ComponentStatusInterface {
 	return newComponentStatuses(c)
 }
 
+// WatchOptions is passed as a parameter to per-resource Watch method.
+type WatchOptions struct {
```

Comment:
@smarterclayton @lavalamp  - can we make 1. separately?

I think that 2-4 should be relatively simple, but 1 definitely requires more work